server/src/main/java/com/skillmagnet/User/UserController.java

```diff
@@ -32,4 +55,47 @@ public ResponseEntity<User> getUser(@RequestParam("username") String username) {
         return new ResponseEntity<>(user, HttpStatus.OK);
     }
 
+    /**
+     * Retrieve a user by id.
+     * @param id - user's unique int id
+     * @return
+     *          Success: Requested User
+     *          Failure: NOT_FOUND
+     */
+    @GetMapping("/user/{id}")
+    public ResponseEntity<User> getUser(@PathVariable("id") int id) {
+        User user = userRepository.findById(id);
+
+        if (user == null) {
+            return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+        }
+
+        return new ResponseEntity<>(user, HttpStatus.OK);
+    }
+
+    /**
+     * Compares login credentials to stored credentials.
+     * @param username - name of user attempting login
+     * @param password - password provided at login (Plaintext)
+     * @return
+     *         Success(Match): "Passwords Match!"
+     *         Success(Mismatch): "Passwords do not match!"
+     *         Failure: NOT_FOUND
+     */
+    @PutMapping("/user/login")
+    public ResponseEntity<String> loginUser(@RequestParam("username") String username,
+                                              @RequestParam("password") String password) {
+        User user = userRepository.findByUsername(username);
+        if (user == null) {
+            return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+        }
+
+        if(user.passwordMatch(password)) {
+            user.setLast_login(LocalDateTime.now());
+            userRepository.save(user);
+            return new ResponseEntity<>("Passwords Match!", HttpStatus.OK);
+        }
+
+        return new ResponseEntity<>("Passwords do not match!", HttpStatus.OK);
```

Comment:
:wrench: I would expect this to throw a 401 Unauthorized if password's dont match. Also, I would make this return the user object as this endpoint is probably the primary spot where the FE will get user info on login.